### ISSUE TITLE
Mutating operation on array not triggering update callback

### ISSUE THREAD

**Issue by Annih** [Annih/chef-updatable-attributes#7]
### Context
- **Cookbook version:** 1.1.0
- **Chef version:** 17.9.46

### How to reproduce
Setup the following code in attributes file:

```ruby
default['array'] = []
default['update_detected'] = false

on_attribute_update('array', init_on_registration: false) do
  default['update_detected'] = true
end

default['array'] << 1
```

### Expected behaviour

The update should be performed and detected, thus the registered block called and the new attributes values should be:
-  `node['array'] == [1]`
-  `node['update_detected'] == true`

### Actual behaviour
The update is properly performed, but the registered block is not called, resulting attributes values are:
-  `node['array'] == [1]`
-  `node['update_detected'] == false`

**Comment by Annih** [comment]
Note that `default['array'] += [1]` is properly handled as it is equivalent to `default['array'] = default['array'] + [1]`; not a mutating operation on the Array.

**Comment by Annih** [comment]
Note the example is using `<<` but any other mutator should be handled the same way: `[]=`, `push`, `append`, `delete`, etc.

**Comment by Annih** [comment]
Issue reported upstream https://github.com/chef/chef/issues/13995 with PR https://github.com/chef/chef/pull/13996

**Comment by Annih** [comment]
Fixed in `v1.1.1`
